app/controllers/application_controller.rb

```diff
@@ -1,7 +1,36 @@
 class ApplicationController < ActionController::Base
   helper_method :get_path
+  helper_method :current_user
+  helper_method :logged_in?
 
   def get_path(path_prefix, options = {})
     return Rails.application.routes.url_helpers.send("#{path_prefix}_path", options)
   end
+
+  def current_user
+    @user ||= User.find_by(id: session[:user_id])
+  end
+
+  def logged_in?
+    current_user.present?
+  end
+
+  # On a controller that follows Rails naming conventions ("UsersController"), this will return the corresponding model ("User")
+  def my_model
```

Comment:
I'm assuming this is to help dry up your controller actions? 

Although I can save some lines of code, to me this is at the cost of clarity e.g. seeing `@user = get_object` makes me a little anxious about what is `get_object` even doing. Then I find the method to make sure no weird side effects are there, but I had to leave the code I was reading to figure out what is happening. Now doing this is quite common for developers, but It can reduce the clarity of the code at the call site. 

It's a little too early for this kind of abstraction. I believe it would be better to wait for more friction before you decide on a solution 

There are also many cases where the controller name and the model name don't map

Reply:
Yep, I was trying to avoid writing out `@user = User.find_by(id: params[:id])` and catching the error with a redirect to the 404 page if it's not found in every controller.

I could just override `my_model` in controllers that don't match the model name when the time comes.

But yeah maybe it's premature DRYing. I get that it makes it less clear, but not sure where to find the balance between readability and not repeating logic.